### PR TITLE
[WIP] codecoverage/bot: Move artifacts and report dirs to a temporary dir

### PR DESCRIPTION
**Pull request by gabriel-v**
Closes https://github.com/mozilla/release-services/issues/1411

It doesn't look like these directories should be needed anywhere else in the code.

### CODE REVIEW

**Comment by gabriel-v** [comment]
Tested these changes locally and they work.

@marco-c some questions:
- For the coverage artifacts, wouldn't we want them to be cached under `$cache_root/ccov-artifacts/<revision_hash>/` instead of `$cache_root/ccov-artifacts/`? Otherwise, if the cache is kept between runs, the job will never download other artifacts (because [existing artifacts are not downloaded again](https://github.com/mozilla/release-services/blob/64a4477e24bc72387246447bff7d39e23f48f221/src/shipit_code_coverage/shipit_code_coverage/taskcluster.py#L77)). In the current implementation where the artifacts are stored in the source dir, I assume that's after a fresh clone, so they are always downloaded and shouldn't be cached. So maybe we should download them in a temporary directory instead?
- For the code coverage reports for wpt, wouldn't a temporary directory work better than something under `$cache_root`?

**Comment by marco-c** [comment]
These artifacts are only needed once (for a single run of the coverage job) or in some cases twice (by the normal coverage job and by the chunk mapping job), so I think we shouldn't cache them. We can move them to another directory, but I wouldn't cache them.

**Comment by gabriel-v** [comment]
Ok, I'll use temporary dirs for both of those.

File: src/codecoverage/bot/code_coverage_bot/codecov.py [hunk 1]
```diff
@@ -38,6 +39,9 @@ def __init__(self, revision, cache_root, client_id, access_token):
 
         assert os.path.isdir(cache_root), 'Cache root {} is not a dir.'.format(cache_root)
         self.repo_dir = os.path.join(cache_root, 'mozilla-central')
+        self.temp_dir = tempfile.mkdtemp()
```
Comment (marco-c) [inline src/codecoverage/bot/code_coverage_bot/codecov.py]: Nit: this could just be a local variable, it's not used anywhere else than in this function

**Pull request merged**